### DISCUSSION TITLE
`clangd` fails to handle preprocessor suddenly

### DISCUSSION THREAD

**Discussion by ChenZhongPu** [LazyVim/LazyVim#620] [Q&A] [answered]
Everything went well till yesterday, but now `clangd` fails to handle preprocessor in C.

```lua
{
    "neovim/nvim-lspconfig",
    opts = {
      servers = {
        clangd = {
          mason = false,
        }, -- for C, C++. Note that `clangd` is shipped with OS.
      },
   },
}
```

I use preprocessor to check AVX and NRON support:

```c
#if defined(__arm_neon__)
#include <arm_neon.h>
#else
#include <immintrin.h>
#endif
```

On my Linux machine which is a x86-64 desktop, only `immintrin.h`  is defined, and the `arm_neon.h`  part is expected to be colored in gray.

Of course, the compile and running behavior are expected, but why is `clangd` in nvim unable to handle it?

![image](https://user-images.githubusercontent.com/2310958/232961256-20d16438-9530-4bdc-a835-82de7edf380b.png)

BTW, I also tested using `kate`, which is able to use `clangd` as the LSP automatically. Its display is expected:

![image](https://user-images.githubusercontent.com/2310958/232961095-86713b10-0a29-4ab3-a388-6cf4065872d0.png)

**Comment by amaanq** [comment, marked as answer]
It's from tokyonight, a recent commit reverted linking lsp comments to comment (which I initially added, dk why it was reverted), but I pred a revert for that a few hrs ago...I totally agree w/ you here that this should be highlighted as `@comment` :)

**Reply by ChenZhongPu** [reply]
Thanks. I decide to switch to `catppuccin`, and now it works again.

**Reply by folke** [reply]
Yeah, needs to be fixed. I totally forgot why that was linked to `@comment`. Will fix
